tests/core/test_test.py

```diff
@@ -256,13 +256,54 @@ async def test_action_unlikely_intent_correctly_predicted(
     )
 
     result = await rasa.core.test.test(
-        str(file_name), agent, out_directory=str(tmp_path),
+        str(file_name), agent, out_directory=str(tmp_path), fail_on_prediction_errors = True
     )
     assert "report" in result.keys()
     assert result["report"]["conversation_accuracy"]["correct"] == 1
     assert result["report"]["conversation_accuracy"]["with_warnings"] == 0
 
 
+async def test_action_unlikely_intent_fail_on_prediction_errors(
+        monkeypatch: MonkeyPatch, tmp_path: Path, moodbot_domain: Domain
+):
+    monkeypatch.setattr(
+        SimplePolicyEnsemble,
+        "probabilities_using_best_policy",
+        _probabilities_with_action_unlikely_intent_for(["mood_unhappy"]),
+    )
```

Comment:
@alwx Isn't `test_action_unlikely_intent_not_found` testing the cases where `action_unlikely_intent` was expected but was not found and hence a prediction error is triggered?

Reply:
Yes, but it checks the report, and this one fails if intent was not found & there is no report that needs to be generated (which happens when `fail_on_prediction_errors=True`). I think it makes sense to have this test because the condition of whether the exception needs to be raised needs to be tested regardless of anything else.